### PR TITLE
Add garbage collector command to launcher

### PR DESCRIPTION
**Pull request by libeanim**
The seafile garbage collector can be started via the `gc` command option of the `launcher`.
This will stop the seafile-server inside the container, then run the `seaf-gc.sh` script and redirect its output to `/var/log/gc.log`.
Afterwards the whole container will be restarted.

It is based on the [gc-cleanup-script](https://manual.seafile.com/maintain/seafile_gc.html#gc-cleanup-script-for-community-version) shown in the seafile documentation.

### CODE REVIEW

**Comment by lins05** [comment]
Thanks, I'll take a look soon.

File: launcher [hunk 1]
```diff
@@ -224,6 +225,25 @@ restart() {
     start
 }
 
+gc() {
+    ensure_container_running
+    (
+        loginfo "Stop seafile and start garbage collector"
+        docker exec -it seafile /bin/bash -c "\
```
Comment (lins05) [inline launcher]: It's better to create a shell script, e.g. `scripts/gc.sh`, and put the command there. The script should exit with the code of the seafile-gc.sh script. The `scripts` dir is mounted to `/scripts` inside the container.

File: launcher [hunk 2]
```diff
@@ -224,6 +225,25 @@ restart() {
...
+            /opt/seafile/seafile-server-latest/seafile.sh stop; \
+            sleep 10; \
+            /opt/seafile/seafile-server-latest/seaf-gc.sh >> /var/log/gc.log; \
```
Comment (lins05) [inline launcher]: Use `tee` instead of simple redirection. It would be useful to show the output of the gc command in real time.

File: launcher [hunk 3]
```diff
@@ -224,6 +225,25 @@ restart() {
...
+            sleep 3"
+        loginfo "Garbage collector finished! Restart container..."
+        running=$(docker ps | awk '{ print $1, $(NF) }' | grep " seafile$" | awk '{ print $1 }' || true)
+        if [[ $running == "" ]]; then
+            start
+        else
+            restart
```
Comment (lins05) [inline launcher]: Why do we need to call start/restart here? We can simply start the server processes again by `seafile.sh start` inside the container.

Comment (libeanim) [inline launcher]: Yeah, wasn't sure whether the seafile-servers PID is tracked and just wanted a clean start.

However, I just took a look at the [watch_controller](https://github.com/haiwen/seafile-docker/blob/master/scripts/start.py#L24) function in `start.py` when a 'gc cleanup' takes longer than 4 * 5 seconds the start.py script exits which would terminate the container.
I need to fix that as well.

**Comment by libeanim** [comment]
Guess I can also get rid of the `sleep` commands because the seafile-server is stopped directly via the shell script and not via a service manager manager?
```
/opt/seafile/seafile-server-latest/seafile-server stop
sleep 10
```

However it doesn't hurt either...

**Comment by lins05** [comment]
Thanks @libeanim for the patch!

**Pull request merged**